cpp/src/arrow/pretty_print.cc

```diff
@@ -143,13 +145,51 @@ class ArrayPrinter : public PrettyPrinter {
   }
 
   template <typename T>
-  inline typename std::enable_if<IsInteger<T>::value, Status>::type WriteDataValues(
-      const T& array) {
+  inline typename std::enable_if<
+      IsInteger<T>::value && !is_date<typename T::TypeClass>::value &&
+          !is_time<typename T::TypeClass>::value &&
+          !std::is_same<TimestampType, typename T::TypeClass>::value,
```

Comment:
`!is_date<typename T::TypeClass>::value` and `!std::is_same<TimestampType, typename T::TypeClass>::value` are needless.